extensions/amp-bind/0.1/amp-state.js

```diff
@@ -207,15 +211,37 @@ export class AmpState extends AMP.BaseElement {
   /**
    * @param {boolean} isInit
    * @param {boolean=} opt_refresh
-   * @return {!Promise<undefined>}
+   * @return {!Promise}
    * @private
    */
   fetchAndUpdate_(isInit, opt_refresh) {
+    // On init, we reuse the deferred created in the constructor.
+    if (!isInit) {
+      this.loadingDeferred_ = new Deferred();
+    }
+
     // Don't fetch in prerender mode.
     return this.getAmpDoc()
       .whenFirstVisible()
       .then(() => this.prepareAndSendFetch_(isInit, opt_refresh))
-      .then(json => this.updateState_(json, isInit));
+      .then(json => this.updateState_(json, isInit))
+      .then(() => {
+        this.loadingDeferred_.resolve();
```

Comment:
This will lead to a race condition if `loadingDeferred` is updated between fetch and resolution.

Reply:
Nice catch!